File: frontend/src/types/task.ts
/** 任务状态 */
export type TaskStatus = 'todo' | 'in_progress' | 'done'

/** 任务类型 */
export type TaskType = 'dev' | 'test' | 'design' | 'research' | 'deploy' | 'training' | 'review' | 'other'

/** 任务信息（对齐后端 TaskVO） */
export interface Task {
  id: number
  projectId: number
  parentTaskId?: number
  requirementId?: number
  promotionUnitId?: number
  title: string
  description?: string
  type: TaskType
  status: TaskStatus
  priority: string
  assignedTo?: number
  plannedStart?: string
  plannedEnd?: string
  actualStart?: string
  actualEnd?: string
  completionRate: number
  wbsCode?: string
  children?: Task[]
  dependsOnTaskIds?: number[]
  createdAt: string
}

/** 任务创建请求 */
export interface TaskCreateRequest {
  title: string
  description?: string
  type?: TaskType
  priority?: string
  assignedTo?: number
  requirementId?: number
  plannedStart?: string
  plannedEnd?: string
}

/** 进度更新请求（对齐后端：completionRate + remark） */
export interface ProgressUpdateRequest {
  completionRate: number
  remark?: string
}

/** 依赖请求 */
export interface DependencyRequest {
  dependsOnTaskId: number
  dependencyType?: 'FS' | 'FF' | 'SS' | 'SF'
}

File: backend/pm-task/src/main/java/com/pm/task/domain/Task.java
package com.pm.task.domain;

import com.baomidou.mybatisplus.annotation.TableName;
import com.pm.common.base.BaseEntity;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.time.LocalDate;

/**
 * 任务实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("task")
@Schema(description = "任务")
public class Task extends BaseEntity {

    @Schema(description = "所属项目ID")
    private Long projectId;

    @Schema(description = "父任务ID（WBS层级）")
    private Long parentTaskId;

    @Schema(description = "关联需求ID")
    private Long requirementId;

    @Schema(description = "关联推广单元ID")
    private Long promotionUnitId;

    @Schema(description = "任务标题")
    private String title;

    @Schema(description = "任务描述")
    private String description;

    @Schema(description = "任务类型：dev/test/deploy/training/design/review")
    private String type;

    @Schema(description = "状态：todo/in_progress/done")
    private String status;

    @Schema(description = "优先级：critical/high/medium/low")
    private String priority;

    @Schema(description = "负责人ID")
    private Long assignedTo;

    @Schema(description = "计划开始日期")
    private LocalDate plannedStart;

    @Schema(description = "计划结束日期")
    private LocalDate plannedEnd;

    @Schema(description = "实际开始日期")
    private LocalDate actualStart;

    @Schema(description = "实际结束日期")
    private LocalDate actualEnd;

    @Schema(description = "完成百分比")
    private BigDecimal completionRate;

    @Schema(description = "预估工时（小时）")
    private BigDecimal estimatedHours;

    @Schema(description = "实际工时（小时）")
    private BigDecimal actualHours;

    @Schema(description = "WBS编码")
    private String wbsCode;

    @Schema(description = "排序序号")
    private Integer sortOrder;
}
